pdfminer/pdffont.py

```diff
@@ -76,7 +78,33 @@ def get_widths2(seq):
 #assert get_widths2([1,2,3,4,5]) == {1:(3, (4,5)), 2:(3, (4,5))}
 #assert get_widths2([1,[2,3,4,5],6,[7,8,9]]) == {1:(2, (3,4)), 6:(7, (8,9))}
 
-
+def computeStat(it):
+	stat=defaultdict(lambda: defaultdict(int))
+	for fd in it:
+		for cat, size in itemz(fd):
+			stat[cat][size]+=1
+	metrics={}
+	for cat, sizes in stat.items():
+		res=max(sizes.items(), key=lambda x: x[1])
+		if res[1] == 1:
+			metrics[cat]=mean(sizes.keys())
+		else:
+			metrics[cat]=res[0]
+	return metrics
+
+def computeFallbackFontMetrics(FONT_METRICS):
+    #TODO:
+    # * weight-dependent metrics
+    # * machine-learning based metrics prediction
+    #weights=defaultdict(list)
+    #for f in FONT_METRICS.values():
+    #    weights[f[0]['FontWeight']].append(f)
+    charMetrics=computeStat((fd[1] for fd in FONT_METRICS.values()))
+    restMetrics=computeStat(({k:v for k,v in fd[0].items() if isinstance(v, (int, float))} for fd in FONT_METRICS.values()))
```

Comment:
If I understand correctly, this filters out the values font metrics that are not `int` or `float`. Why is that? 

This skips FontName, which should be in the font description according to pdf reference. Also see Table 5.19 from the [pdf reference](https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/pdf_reference_1-7.pdf).

Reply:
It computes mean. For non-numerical we can compute the most frequent, but it is not needed, we currently need only sizes in order to glue the chars into words and lines correctly.